File: solutions/c#/ParkingLot/ParkingLotDemo.cs
using ParkingLot.VehicleTypes;

namespace ParkingLot
{
    public class ParkingLotDemo
    {
        public static void Run()
        {
            ParkingLot parkingLot = ParkingLot.GetInstance();
            parkingLot.AddLevel(new Level(1, 100));
            parkingLot.AddLevel(new Level(2, 80));

            Vehicle car = new Car("ABC123");
            Vehicle truck = new Truck("XYZ789");
            Vehicle motorcycle = new Motorcycle("M1234");

            // Park vehicles
            parkingLot.ParkVehicle(car);
            parkingLot.ParkVehicle(truck);
            parkingLot.ParkVehicle(motorcycle);

            // Display availability
            parkingLot.DisplayAvailability();

            // Unpark vehicle
            parkingLot.UnparkVehicle(motorcycle);

            // Display updated availability
            parkingLot.DisplayAvailability();
        }
    }
}

File: solutions/golang/parkinglot/parking_lot_demo.go
package parkinglot

func Run() {
	parkingLot := GetParkingLotInstance()
	parkingLot.AddLevel(NewLevel(1, 100))
	parkingLot.AddLevel(NewLevel(2, 80))

	car := NewCar("ABC123")
	truck := NewTruck("XYZ789")
	motorcycle := NewMotorcycle("M1234")

	parkingLot.ParkVehicle(car)
	parkingLot.ParkVehicle(truck)
	parkingLot.ParkVehicle(motorcycle)

	parkingLot.DisplayAvailability()

	parkingLot.UnparkVehicle(motorcycle)

	parkingLot.DisplayAvailability()
}
